Mathlib/Algebra/Star/Order.lean

```diff
@@ -234,11 +234,24 @@ lemma IsSelfAdjoint.mono {x y : R} (h : x ≤ y) (hx : IsSelfAdjoint x) : IsSelf
 lemma IsSelfAdjoint.of_nonneg {x : R} (hx : 0 ≤ x) : IsSelfAdjoint x :=
   (isSelfAdjoint_zero R).mono hx
 
+theorem conjugate_lt_conjugate {a b : R} (hab : a < b) {c : R} (hc : IsRegular c) :
+    star c * a * c < star c * b * c := by
+  rw [(conjugate_le_conjugate hab.le _).lt_iff_ne, hc.right.ne_iff, hc.star.left.ne_iff]
+  exact hab.ne
+
+theorem conjugate_lt_conjugate' {a b : R} (hab : a < b) {c : R} (hc : IsRegular c) :
+    c * a * star c < c * b * star c := by
+  simpa only [star_star] using conjugate_lt_conjugate hab hc.star
+
+theorem conjugate_pos {a : R} (ha : 0 < a) {c : R} (hc : IsRegular c) : 0 < star c * a * c := by
+  simpa only [mul_zero, zero_mul] using conjugate_lt_conjugate ha hc
```

Comment:
> Am I right that in this generality you could get the related but different result where the hypothesis on a is nonnegativity and regularity and the conclusion is regularity?
>
> If so, do you mind adding it?

I'm not really sure how that would follow; can you give the statement you're thinking of, once you're not on mobile?

Reply:
The statement I was originally thinking of is trivially true, and doesn't require positivity:
```lean
import Mathlib.Algebra.Star.Order

variable {R : Type*} [NonUnitalSemiring R] [PartialOrder R] [StarRing R] [StarOrderedRing R]

example {a : R} (ha : 0 ≤ a) (ha' : IsRegular a) {c : R} (hc : IsRegular c) : IsRegular (star c * a * c) :=
  isRegular_mul_and_mul_iff.mpr ⟨(isRegular_mul_and_mul_iff.mpr ⟨hc.star, ha'⟩).1, hc⟩ |>.1
```
But I believe what I was actually wanting is something more along the lines of: if `a : R` is left-regular, then `star a * a` is regular.

However, even though it's true in C⋆-algebras (easy to see but complicated proof: represent the C⋆-algebra on a Hilbert space, notice that left-regularity corresponds to injectivity, and recall that `star a * a` and `a` have the same kernel; then `star a * a` is self-adjoint and left-regular, hence regular), I don't see how to get it in this generality. With it, one would hope to generalize my example above to `IsLeftRegular c` (and that seems hopeless too because we can't actually write `a = star b * b` because of our definition of `StarOrderedRing`), but oh well.

In any case, unless you see something I don't with the above, we'll just skip it for this PR.